EDA

Problem Statement:

to identify the key factors that influence loan approval decisions and to understand whether customer attributes such as income, credit score, assets, and dependents can be used to predict loan approval outcomes.


In [1]:
import pandas as pd

In [2]:
df=pd.read_csv("Cleaned_dataset.csv")

Target Column : loan_status

Strong Predictors : 

loan_amount

assests_columns

credit_score

income_annum

In [3]:
df

loan_id  no_of_dependents      education self_employed  income_annum  \
0           1                 2       Graduate            No       9600000   
1           2                 0   Not Graduate           Yes       4100000   
2           3                 3       Graduate            No       9100000   
3           4                 3       Graduate            No       8200000   
4           5                 5   Not Graduate           Yes       9800000   
...       ...               ...            ...           ...           ...   
4264     4265                 5       Graduate           Yes       1000000   
4265     4266                 0   Not Graduate           Yes       3300000   
4266     4267                 2   Not Graduate            No       6500000   
4267     4268                 1   Not Graduate            No       4100000   
4268     4269                 1       Graduate            No       9200000   

      loan_amount  loan_term  cibil_score  residential_assets_value  \
0        29900000         12          778                   2400000   
1        12200000          8          417                   2700000   
2        29700000         20          506                   7100000   
3        30700000          8          467                  18200000   
4        24200000         20          382                  12400000   
...           ...        ...          ...                       ...   
4264      2300000         12          317                   2800000   
4265     11300000         20          559                   4200000   
4266     23900000         18          457                   1200000   
4267     12800000          8          780                   8200000   
4268     29700000         10          607                  17800000   

      commercial_assets_value  luxury_assets_value  bank_asset_value  \
0                    17600000             22700000           8000000   
1                     2200000              8800000           3300000   
2                     4500000             33300000          12800000   
3                     3300000             23300000           7900000   
4                     8200000             29400000           5000000   
...                       ...                  ...               ...   
4264                   500000              3300000            800000   
4265                  2900000             11000000           1900000   
4266                 12400000             18100000           7300000   
4267                   700000             14100000           5800000   
4268                 11800000             35700000          12000000   

      loan_status  total_assets  
0               1      50700000  
1               0      17000000  
2               0      57700000  
3               0      52700000  
4               0      55000000  
...           ...           ...  
4264            0       7400000  
4265            1      20000000  
4266            0      39000000  
4267            1      28800000  
4268            1      77300000  

[4269 rows x 14 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4269 entries, 0 to 4268
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   loan_id                   4269 non-null   int64 
 1   no_of_dependents          4269 non-null   int64 
 2   education                 4269 non-null   object
 3   self_employed             4269 non-null   object
 4   income_annum              4269 non-null   int64 
 5   loan_amount               4269 non-null   int64 
 6   loan_term                 4269 non-null   int64 
 7   cibil_score               4269 non-null   int64 
 8   residential_assets_value  4269 non-null   int64 
 9   commercial_assets_value   4269 non-null   int64 
 10  luxury_assets_value       4269 non-null   int64 
 11  bank_asset_value          4269 non-null   int64 
 12  loan_status               4269 non-null   int64 
 13  total_assets              4269 non-null   int64 
dtypes: int64(12), object(2)


In [5]:
df.columns

Index(['loan_id', 'no_of_dependents', 'education', 'self_employed',
       'income_annum', 'loan_amount', 'loan_term', 'cibil_score',
       'residential_assets_value', 'commercial_assets_value',
       'luxury_assets_value', 'bank_asset_value', 'loan_status',
       'total_assets'],
      dtype='object')

In [6]:
df["loan_status"].value_counts()

loan_status
1    2656
0    1613
Name: count, dtype: int64

# Cibil Score Vs Loan Status

In [7]:
df.groupby('loan_status')['cibil_score'].mean()


loan_status
0    429.468072
1    703.461973
Name: cibil_score, dtype: float64

Applicant with higher cibil score get their loan approved and rest does not.

On an Average of 703 cibil score Applicant Loan is approved and below 430 the loan is not approved.

Approved loans have significantly higher average CIBIL scores, confirming credit history as the primary driver of loan approval decisions.

# Income-Loan Ratio vs Loan Status

In [8]:
df["income_loan_ratio"]=df["income_annum"]/df["loan_amount"]

In [9]:
df.head()

loan_id  no_of_dependents      education self_employed  income_annum  \
0        1                 2       Graduate            No       9600000   
1        2                 0   Not Graduate           Yes       4100000   
2        3                 3       Graduate            No       9100000   
3        4                 3       Graduate            No       8200000   
4        5                 5   Not Graduate           Yes       9800000   

   loan_amount  loan_term  cibil_score  residential_assets_value  \
0     29900000         12          778                   2400000   
1     12200000          8          417                   2700000   
2     29700000         20          506                   7100000   
3     30700000          8          467                  18200000   
4     24200000         20          382                  12400000   

   commercial_assets_value  luxury_assets_value  bank_asset_value  \
0                 17600000             22700000           8000000   
1                  2200000              8800000           3300000   
2                  4500000             33300000          12800000   
3                  3300000             23300000           7900000   
4                  8200000             29400000           5000000   

   loan_status  total_assets  income_loan_ratio  
0            1      50700000           0.321070  
1            0      17000000           0.336066  
2            0      57700000           0.306397  
3            0      52700000           0.267101  
4            0      55000000           0.404959

In [10]:
df.describe()

loan_id  no_of_dependents  income_annum   loan_amount    loan_term  \
count  4269.000000       4269.000000  4.269000e+03  4.269000e+03  4269.000000   
mean   2135.000000          2.498712  5.059124e+06  1.513345e+07    10.900445   
std    1232.498479          1.695910  2.806840e+06  9.043363e+06     5.709187   
min       1.000000          0.000000  2.000000e+05  3.000000e+05     2.000000   
25%    1068.000000          1.000000  2.700000e+06  7.700000e+06     6.000000   
50%    2135.000000          3.000000  5.100000e+06  1.450000e+07    10.000000   
75%    3202.000000          4.000000  7.500000e+06  2.150000e+07    16.000000   
max    4269.000000          5.000000  9.900000e+06  3.950000e+07    20.000000   

       cibil_score  residential_assets_value  commercial_assets_value  \
count  4269.000000              4.269000e+03             4.269000e+03   
mean    599.936051              7.472617e+06             4.973155e+06   
std     172.430401              6.503637e+06             4.388966e+06   
min     300.000000             -1.000000e+05             0.000000e+00   
25%     453.000000              2.200000e+06             1.300000e+06   
50%     600.000000              5.600000e+06             3.700000e+06   
75%     748.000000              1.130000e+07             7.600000e+06   
max     900.000000              2.910000e+07             1.940000e+07   

       luxury_assets_value  bank_asset_value  loan_status  total_assets  \
count         4.269000e+03      4.269000e+03  4269.000000  4.269000e+03   
mean          1.512631e+07      4.976692e+06     0.622160  3.254877e+07   
std           9.103754e+06      3.250185e+06     0.484904  1.950656e+07   
min           3.000000e+05      0.000000e+00     0.000000  4.000000e+05   
25%           7.500000e+06      2.300000e+06     0.000000  1.630000e+07   
50%           1.460000e+07      4.600000e+06     1.000000  3.150000e+07   
75%           2.170000e+07      7.100000e+06     1.000000  4.720000e+07   
max           3.920000e+07      1.470000e+07     1.000000  9.070000e+07   

       income_loan_ratio  
count        4269.000000  
mean            0.349534  
std             0.074208  
min             0.250000  
25%             0.285714  
50%             0.333333  
75%             0.405797  
max             0.666667

In [11]:
df.groupby("loan_status")["income_loan_ratio"].mean()

loan_status
0    0.357774
1    0.344530
Name: income_loan_ratio, dtype: float64

no much difference


The income-to-loan ratio shows very minimal varaition between approved and rejected loans, indicating that income alone does not strongly determine approval. It acts as a secondary validation check when primarys are done.

# Total Assets Vs Loan Status

In [12]:
df.groupby("loan_status")["total_assets"].median()

loan_status
0    31900000.0
1    31000000.0
Name: total_assets, dtype: float64

The difference is only of 9lakh which is a very small difference. 

So Total assets alone are not a strong predictor for loan approval

The Median of total assets differ marginally between approved and rejected applicants, suggesting that assets ownership alone does not guarantee loan approval. It can considered as a supporting factor but not the primary one.

In [13]:
df.columns

Index(['loan_id', 'no_of_dependents', 'education', 'self_employed',
       'income_annum', 'loan_amount', 'loan_term', 'cibil_score',
       'residential_assets_value', 'commercial_assets_value',
       'luxury_assets_value', 'bank_asset_value', 'loan_status',
       'total_assets', 'income_loan_ratio'],
      dtype='object')

# No. of Dependencies Vs Loan Status

In [14]:
df.groupby("loan_status")["no_of_dependents"].median()

loan_status
0    3.0
1    2.5
Name: no_of_dependents, dtype: float64

In [15]:
df.groupby("loan_status")["no_of_dependents"].mean()

loan_status
0    2.538128
1    2.474774
Name: no_of_dependents, dtype: float64

Both mean and median values of dependents are similar across loan outcomes, indicating that family size has limited direct impact on approval and is not a decisive factor.

In [16]:
df["loan_amount"].describe()

count    4.269000e+03
mean     1.513345e+07
std      9.043363e+06
min      3.000000e+05
25%      7.700000e+06
50%      1.450000e+07
75%      2.150000e+07
max      3.950000e+07
Name: loan_amount, dtype: float64

In [17]:
df

loan_id  no_of_dependents      education self_employed  income_annum  \
0           1                 2       Graduate            No       9600000   
1           2                 0   Not Graduate           Yes       4100000   
2           3                 3       Graduate            No       9100000   
3           4                 3       Graduate            No       8200000   
4           5                 5   Not Graduate           Yes       9800000   
...       ...               ...            ...           ...           ...   
4264     4265                 5       Graduate           Yes       1000000   
4265     4266                 0   Not Graduate           Yes       3300000   
4266     4267                 2   Not Graduate            No       6500000   
4267     4268                 1   Not Graduate            No       4100000   
4268     4269                 1       Graduate            No       9200000   

      loan_amount  loan_term  cibil_score  residential_assets_value  \
0        29900000         12          778                   2400000   
1        12200000          8          417                   2700000   
2        29700000         20          506                   7100000   
3        30700000          8          467                  18200000   
4        24200000         20          382                  12400000   
...           ...        ...          ...                       ...   
4264      2300000         12          317                   2800000   
4265     11300000         20          559                   4200000   
4266     23900000         18          457                   1200000   
4267     12800000          8          780                   8200000   
4268     29700000         10          607                  17800000   

      commercial_assets_value  luxury_assets_value  bank_asset_value  \
0                    17600000             22700000           8000000   
1                     2200000              8800000           3300000   
2                     4500000             33300000          12800000   
3                     3300000             23300000           7900000   
4                     8200000             29400000           5000000   
...                       ...                  ...               ...   
4264                   500000              3300000            800000   
4265                  2900000             11000000           1900000   
4266                 12400000             18100000           7300000   
4267                   700000             14100000           5800000   
4268                 11800000             35700000          12000000   

      loan_status  total_assets  income_loan_ratio  
0               1      50700000           0.321070  
1               0      17000000           0.336066  
2               0      57700000           0.306397  
3               0      52700000           0.267101  
4               0      55000000           0.404959  
...           ...           ...                ...  
4264            0       7400000           0.434783  
4265            1      20000000           0.292035  
4266            0      39000000           0.271967  
4267            1      28800000           0.320312  
4268            1      77300000           0.309764  

[4269 rows x 15 columns]

# Education Vs Loan Status

In [18]:
pd.crosstab(df['education'], df['loan_status'], normalize=True)*100


loan_status           0          1
education                         
Graduate      18.856875  31.365659
Not Graduate  18.927149  30.850316

In [19]:
pd.crosstab(df['education'], df['loan_status'])


loan_status     0     1
education              
Graduate      805  1339
Not Graduate  808  1317

Cross Table analysis shows there is not much difference but applicants with higher qualication have relatively better approval rates, implies education may indirectly influence approvals 

In [20]:
pd.crosstab(df["self_employed"],df["loan_status"], normalize=True)*100

loan_status            0          1
self_employed                      
No             18.763176  30.873741
Yes            19.020848  31.342235

In [21]:
pd.crosstab(df["self_employed"],df["loan_status"])

loan_status      0     1
self_employed           
No             801  1318
Yes            812  1338

Salaried applicants show relatively higher approval proportions compared to self-employed customers, reflecting banks’ preference for stable and predictable income sources rather than variable earnings.

# Final Insight

Loan approval decisions are primarily driven by credit score, with income, assets, education, employment type, and dependents acting as supporting factors rather than independent decision-makers. This suggests a credit-first risk assessment approach, where additional attributes are used to validate eligibility rather than determine approval on it.

In [22]:
df

loan_id  no_of_dependents      education self_employed  income_annum  \
0           1                 2       Graduate            No       9600000   
1           2                 0   Not Graduate           Yes       4100000   
2           3                 3       Graduate            No       9100000   
3           4                 3       Graduate            No       8200000   
4           5                 5   Not Graduate           Yes       9800000   
...       ...               ...            ...           ...           ...   
4264     4265                 5       Graduate           Yes       1000000   
4265     4266                 0   Not Graduate           Yes       3300000   
4266     4267                 2   Not Graduate            No       6500000   
4267     4268                 1   Not Graduate            No       4100000   
4268     4269                 1       Graduate            No       9200000   

      loan_amount  loan_term  cibil_score  residential_assets_value  \
0        29900000         12          778                   2400000   
1        12200000          8          417                   2700000   
2        29700000         20          506                   7100000   
3        30700000          8          467                  18200000   
4        24200000         20          382                  12400000   
...           ...        ...          ...                       ...   
4264      2300000         12          317                   2800000   
4265     11300000         20          559                   4200000   
4266     23900000         18          457                   1200000   
4267     12800000          8          780                   8200000   
4268     29700000         10          607                  17800000   

      commercial_assets_value  luxury_assets_value  bank_asset_value  \
0                    17600000             22700000           8000000   
1                     2200000              8800000           3300000   
2                     4500000             33300000          12800000   
3                     3300000             23300000           7900000   
4                     8200000             29400000           5000000   
...                       ...                  ...               ...   
4264                   500000              3300000            800000   
4265                  2900000             11000000           1900000   
4266                 12400000             18100000           7300000   
4267                   700000             14100000           5800000   
4268                 11800000             35700000          12000000   

      loan_status  total_assets  income_loan_ratio  
0               1      50700000           0.321070  
1               0      17000000           0.336066  
2               0      57700000           0.306397  
3               0      52700000           0.267101  
4               0      55000000           0.404959  
...           ...           ...                ...  
4264            0       7400000           0.434783  
4265            1      20000000           0.292035  
4266            0      39000000           0.271967  
4267            1      28800000           0.320312  
4268            1      77300000           0.309764  

[4269 rows x 15 columns]